lua/luasnip/nodes/snippet.lua

```diff
@@ -385,20 +464,177 @@ local function _S(snip, nodes, opts)
+---  - `snippet`: The expanding snippet object
+---  - `line_to_cursor`: The line up to the cursor.
+---  - `matched_trigger`: The fully matched trigger (can be retrieved
+---    from `line_to_cursor`, but we already have that info here :D)
+---  - `captures`: Captures as returned by `trigEngine`.
+---
+---  This function will be evaluated in `Snippet:matches()` to decide whether
+---  the snippet can be expanded or not.
+---  Returns a table if the snippet can be expanded, `nil` if can not.
+---
+---  If any field in the returned table is `nil`, the default is used (`trigger` and `captures` as
+---  returned by `trigEngine`, `clear_region` such that exactly the trigger is
+---  deleted, no overridden environment-variables).
+---
+---  A good example for the usage of `resolveExpandParams` can be found in the
+---  implementation of [`postfix`](https://github.com/L3MON4D3/LuaSnip/blob/master/lua/luasnip/extras/postfix.lua).
+---
+---@field condition? LuaSnip.SnipContext.Condition
+---  - `line_to_cursor`: the line up to the cursor.
+---  - `matched_trigger`: the fully matched trigger (can be retrieved
+---    from `line_to_cursor`, but we already have that info here :D).
+---  - `captures`: if the trigger is pattern, contains the capture-groups.
+---    Again, could be computed from `line_to_cursor`, but we already did so.
+---
+---  This function can prevent manual snippet expansion via `ls.expand()`.
+---  Return `true` to allow expansion, and `false` to prevent it.
+---
+---@field show_condition? LuaSnip.SnipContext.ShowCondition
+---  This function is (should be) evaluated by completion engines, indicating
+---  whether the snippet should be included in current completion candidates.
+---  Defaults to a function returning `true`.
+---
+---  This is different from `condition` because `condition` is evaluated by
+---  LuaSnip on snippet expansion (and thus has access to the matched trigger and
+---  captures), while `show_condition` is (should be) evaluated by the
+---  completion engines when scanning for available snippet candidates.
+---
+---@field filetype? string The filetype of the snippet.
+---  This overrides the filetype the snippet is added (via `add_snippet`) as.
+
+---@class LuaSnip.Opts.Snippet: LuaSnip.Opts.SnippetNode
+---@field stored? {[string]: LuaSnip.Node}
+
+---@param context string|LuaSnip.SnipContext The snippet context.
+---  Passing a string is equivalent to passing `{ trig = <the string> }`.
+---@param nodes LuaSnip.Node|LuaSnip.Node[] The nodes that make up the snippet.
+---@param opts? LuaSnip.Opts.Snippet
+---@return LuaSnip.Snippet
 local function S(context, nodes, opts)
 	opts = opts or {}
 
 	local snip = init_snippet_context(node_util.wrap_context(context), opts)
-	snip = vim.tbl_extend("error", snip, init_snippet_opts(opts))
+	local snip = vim.tbl_extend("error", snip, init_snippet_opts(opts))
 
-	snip = _S(snip, nodes, opts)
+	local snip = _S(snip, nodes, opts)
```

Comment:
Why the re-definitions? I think we should have different names if they are necessary

Reply:
Ahh alright, that makes sense.
I think the first one could be `snip_with_opts` instead of `snip`

Reply:
Done in 6204bbd3e99526f766bd029773c5f19145b25ef8